generate-endpoint-scan-pdf-report/generate_ep_scan.py

```diff
@@ -0,0 +1,147 @@
+import traceback
+from datetime import datetime
+
+import jinja2
+import pdfkit
+from intezer_sdk import api
+from intezer_sdk.endpoint_analysis import EndpointAnalysis
+
+
+def format_analysis_time(scan_start: str, scan_end: str):
+    date_format = '%a, %d %b %Y %H:%M:%S %Z'
+    # Calculating the time the endpoint scan took
+    start_date = datetime.strptime(scan_start, date_format)
+    end_date = datetime.strptime(scan_end, date_format)
+    calc_scan_time = end_date - start_date
+    # Setting the time format
+    hours = int(calc_scan_time.total_seconds()) // 3600
+    minutes = int(calc_scan_time.total_seconds() % 3600) // 60
+    seconds = int(calc_scan_time.total_seconds()) % 60
+
+    formatted_time = ''
+    if hours > 0:
+        formatted_time += f"{hours} hour{'s' if hours > 1 else ''},"
+    if minutes > 0:
+        formatted_time += f" {minutes} minute{'s' if minutes > 1 else ''} and "
+
+    formatted_time += f"{seconds} second{'s' if seconds > 1 else ''}"
+    return formatted_time
+
+
+def generate_report(
+        endpoint_analysis: EndpointAnalysis,
+        css_input: str,
+        template_text: str,
+        logo_base64: str,
+        save_html_to_file: bool = False):
+
+    all_sub_analyses = endpoint_analysis.get_sub_analyses()
+    sub_analyses = [sub_analysis for sub_analysis in all_sub_analyses if sub_analysis.verdict == 'malicious']
+
+    if not sub_analyses:
+        sub_analyses = [sub_analysis for sub_analysis in all_sub_analyses if
+                        sub_analysis.verdict in ('suspicious', 'unknown')]
+        sub_analyses.sort(key=lambda value: (value.verdict != 'suspicious', value.verdict))
+        sub_analyses = sub_analyses[:100]
+
+    family_info = []
+    for analysis in sub_analyses:
+        if analysis.verdict != 'malicious':
+            continue
+
+        if analysis.code_reuse is not None:
+            # getting the malicious family name
+            malware_families = []
+            families = sorted(analysis.code_reuse.get('families', []),
+                              key=lambda family_: family_.get('reused_gene_count'), reverse=True)
+
+            for family in families:
+                if family.get('family_type') == 'malware':
+                    malware_families.append(family)
+
+            if malware_families:
+                family_info.append({
+                    'family_name': malware_families[0].get('family_name'),
+                    'family_id': malware_families[0].get('family_id')
+                })
+
+    # Basic info
+    ep_analysis_metadata = endpoint_analysis.result()
+    ep_analysis_metadata['status'] = endpoint_analysis.status.value
+
+    ep_analysis_metadata['calc_scan_time'] = format_analysis_time(ep_analysis_metadata['scan_start_time'],
+                                                                  ep_analysis_metadata['scan_end_time'])
+
+    if ep_analysis_metadata['families'] is not None and len(ep_analysis_metadata['families']) > 0:
```

Comment:
you can just check `ep_analysis_metadata['families']` - in python, empty list or dict are translated to false as well